restapi.go

```diff
@@ -2530,11 +2449,19 @@ func (s *Session) ApplicationCommands(appID, guildID string) (cmd []*Application
 // appID       : The application ID.
 // interaction : Interaction instance.
 // resp        : Response message data.
-func (s *Session) InteractionRespond(interaction *Interaction, resp *InteractionResponse) error {
+func (s *Session) InteractionRespond(interaction *Interaction, resp *InteractionResponse) (err error) {
 	endpoint := EndpointInteractionResponse(interaction.ID, interaction.Token)
 
-	_, err := s.RequestWithBucketID("POST", endpoint, *resp, endpoint)
+	if len(resp.Files) > 0 {
+		contentType, body, err := EncodeWithFiles(interaction, resp.Files)
```

Comment:
Are you sure that the first argument is meant to be `interaction`, not `resp`? I haven't been able to get responding to an interaction with a file working.

Reply:
You may be right, its been a bit since I wrote this. I'll do further testing later today and make sure responding with the callback works (and fix it if it does not) as most of my testing has been with an outgoing webhook instead of the gateway.

Reply:
changed interaction -> resp. and added a working example to examples/slash_commands.